### PR TITLE
Add provider releases page

### PR DESCRIPTION
**Pull request by jeanduplessis**
Adds a releases page for the Upbound Official providers.

### CODE REVIEW

**Comment by jeanduplessis** [comment]
@plumbis @shanecmiller23 would appreciate a review: https://vercel.live/open-feedback/docs-git-provider-releases-upboundio.vercel.app

**Comment by plumbis** [comment]
I'm all for removing or disabling that rule. Not sure we've used a date anywhere yet to know it's a bad rule.

With a monolithic page like this do you expect old releases are removed eventually? 

In #138 I have each update as a unique page and the index page pulls the intro.

https://github.com/upbound/docs/pull/138/files#diff-73018abd514c59ce778073235a8a223cb2246868eed003627086767a3461473fR3

**Comment by jeanduplessis** [comment]
@plumbis 
>With a monolithic page like this do you expect old releases are removed eventually?
I haven't given this prior thought. Your approach is definitely more scalable and one we would have to adopt to prevent an infinitely long page.

I'm biased to making ahead with the single page for now and then switching to multiple pages (one per release) in phase two. I'll open an issue for tracking that and will tackle it soon.

**Comment by jeanduplessis** [comment]
https://github.com/upbound/docs/issues/190

**Review by plumbis — APPROVED** [review]

File: content/providers/releases.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+The below release notes are for the Upbound Official providers. The notes here
+only contain noteworthy changes and you should refer to each release's GitHub
+release notes for full details.
+
+For more information on the release cadence and support protocol refer to the
+provider [support and maintenance]({{<ref "support" >}}) page.
+
+## 2023-06-27
+
+### `upbound/provider-aws` v0.37.0
```
[the remark this answers is not in the document]
Comment (jeanduplessis) [reply content/providers/releases.md:15]: 🤔 Possibly, yes. I'm going to add this consideration to #190 for consideration there.

**Comment by jeanduplessis** [comment]
@shanecmiller23 I'm taking a bias to action and going ahead with what we have here for now. We should definitely consider design aesthetics when I tackle #190

**Pull request merged**